### PR TITLE
[ci] Make some CI jobs green

### PR DESCRIPTION
**Pull request by tsapeta**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -323,9 +323,6 @@ jobs:
       - update_submodules
       - run: git lfs pull
       - fetch_cocoapods_specs
-      - yarn_install:
-          working_directory: ~/project/tools-public
-      - run: ~/project/tools-public/generate-files-ios.js
```
Comment (sjchmiela) [inline .circleci/config.yml]: Those are needed

File: .circleci/config.yml [hunk 2]
```diff
@@ -517,4 +511,4 @@ jobs:
       - decrypt_secrets_if_possible
       - run:
           working_directory: ~/expo/apps/test-suite
-          command: ./publish.sh
+          command: nix run expo.procps --command ./publish.sh
```
Comment (sjchmiela) [inline .circleci/config.yml]: The script starts with
```sh
#!/usr/bin/env nix-shell
#!nix-shell -I nixpkgs=../../nix --packages direnv procps git yarn -i "direnv exec . bash"
```

Doesn't this suffice, @nicknovitski?

File: android/expoview/build.gradle [hunk 3]
```diff
@@ -97,17 +97,20 @@ android {
...
-  workingDir '../../tools/expotools/bin'
-
-  if (Os.isFamily(Os.FAMILY_WINDOWS)) {
-    commandLine 'cmd.exe', '/c', "expotools.js android-generate-dynamic-macros --configuration ${configuration}"
-  } else {
-    // NOTE(brentvatne): this is a common install path for node
-    environment 'PATH', "${environment.PATH}:/usr/local/bin"
-    commandLine "./expotools.js", "android-generate-dynamic-macros", "--configuration", configuration
+task generateDynamicMacros {
+  doLast {
```
Comment (sjchmiela) [inline android/expoview/build.gradle]: I guess it moves the task from configuration phase to build phase? Wouldn't we want to do this when configuring? Eg. to have up-to-date `BasePackageList` after configuring the project?

**Review by sjchmiela — CHANGES_REQUESTED** [review]
🤔

File: .circleci/config.yml (hunk 2 above)
Comment (tsapeta) [reply .circleci/config.yml]: Reverted this change, just wanted to check if that could help. Anyway, seems like `expo publish` command run by this script cannot spawn `ps` process 🤔 

```
...
[13:07:11] Published
[13:07:11] Your URL is

https://exp.host/@exponent_ci_bot/test-suite?release-channel=1f0242223f98c392681a3cbf3e5e4597c79373d6

Error: spawn ps ENOENT
    at Process.ChildProcess._handle.onexit (internal/child_process.js:240:19)
    at onErrorNT (internal/child_process.js:415:16)
    at process._tickCallback (internal/process/next_tick.js:63:19)
error Command failed with exit code 1.
```

Comment (nicknovitski) [reply .circleci/config.yml]: @sjchmiela I agree, only one or the other should be necessary.  But neither are working?

File: .circleci/config.yml [hunk 4]
```diff
@@ -338,9 +336,6 @@ jobs:
       - update_submodules
       - run: git lfs pull
       - fetch_cocoapods_specs
-      - yarn_install:
-          working_directory: ~/project/tools-public
-      - run: ~/project/tools-public/generate-files-ios.js
```
Comment (sjchmiela) [inline .circleci/config.yml]: You sure we don't have to generate files before building Expo Client?

File: android/expoview/build.gradle [hunk 5]
```diff
@@ -105,8 +105,6 @@ task generateDynamicMacros(type: Exec) {
   if (Os.isFamily(Os.FAMILY_WINDOWS)) {
     commandLine 'cmd.exe', '/c', "expotools.js android-generate-dynamic-macros --configuration ${configuration}"
   } else {
-    // NOTE(brentvatne): this is a common install path for node
-    environment 'PATH', "${environment.PATH}:/usr/local/bin"
```
Comment (sjchmiela) [inline android/expoview/build.gradle]: Is @brentvatne ok with this change?

Comment (tsapeta) [reply android/expoview/build.gradle]: since we use nix, I think we no longer need to override PATH here

**Review by sjchmiela — APPROVED** [review]
LGTM!

**Pull request merged**